ethereumj-core/src/main/java/org/ethereum/crypto/jce/SpongyCastleProvider.java

```diff
@@ -3,26 +3,11 @@
 import org.spongycastle.jce.provider.BouncyCastleProvider;
 
 import java.security.Provider;
-import java.security.Security;
 
 public final class SpongyCastleProvider {
 
-  public static final String PROVIDER_NAME = BouncyCastleProvider.PROVIDER_NAME;
-
   private static class Holder {
-    private static final BouncyCastleProvider INSTANCE;
-
-    static {
-      Provider provider = Security.getProvider(PROVIDER_NAME);
-
-      if (provider == null) {
-        INSTANCE = new BouncyCastleProvider();
-        Security.addProvider(INSTANCE);
-      } else {
-        assert provider instanceof BouncyCastleProvider;
-        INSTANCE = (BouncyCastleProvider) provider;
-      }
-    }
+    private static final Provider INSTANCE = new BouncyCastleProvider();
```

Comment:
Hey @Nashatyrev, this was originally my addition.

We realized later that this pattern of dynamically appending a Java Security Provider causes issues in a JVM environment with hot reloading (dev environments for web development, for example).

`Security.addProvider` adds the object instance to some global lookup table. This results in a space / classloader leak when doing class reloading as a reference to the original classloader is kept, and ClassCastExceptions occur in ECKey code because instances from the SpongyCastle library are crossing classloader instances. This is because `Security.getProvider` will return the instance that came from the original classloader.

I removed the code that interacts with the security provider lookup table, and there is no detriment to EthereumJ. There is no code path where a specific security provider is looked up by string name, rather than explicitly passing an instance of a Provider.